JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/GroupBean.java

```diff
@@ -8,132 +11,119 @@
                 .getResultList();
 
 
         Map<String, String> mappedUserGroup = new HashMap<String, String>(usergroup.size());
-        UserGroup[] arrayOfUserGroup = (UserGroup[]) usergroup.toArray();
+        //UserGroup[] arrayOfUserGroup = (UserGroup[]) usergroup.toArray();
 
-        for (int i = 0;i < usergroup.size(); i++) {
+        for (UserGroup ug: usergroup) {
 
-            UserGroup userGroup = arrayOfUserGroup[i];//how to operate with user list????
-           /* String nickName = (String)em.createQuery("SELECT distinct u.nickName FROM UserGroup p JOIN User u WHERE p.userId =:userid ")
-                    .setParameter("userid", userGroup.getUserId())
+            // UserGroup userGroup = arrayOfUserGroup[i];
+            User nickName = (User)em.createQuery("SELECT u.member FROM UserGroup u WHERE u.member.userId =:userid")
+                    .setParameter("userid", ug.getUserId())
                     .getSingleResult();
-            String access =   (String)em.createQuery("SELECT distinct a.accessName FROM UserGroup u JOIN u.access a WHERE u.userid")
-                    .setParameter("userid", userGroup.getUserId())
+            Access access =   (Access)em.createQuery("SELECT u.access FROM UserGroup u WHERE u.access.accessId =:accessid")
+                    .setParameter("accessid", ug.getAccess().getAccessId())
                     .getSingleResult();
-             mappedUserGroup.put(nickName, access);
-*/
+            mappedUserGroup.put(nickName.getNickName(), access.getAccessName());
+
         }
         return mappedUserGroup;
 
     }
 
 
-    public void updateMember(int groupId, String userNickName, String accessRights) {
-
+/*    public void updateMember(int groupId, String userNickName, String accessRights) {
+//do we update usernickname also?
 
         int userId = getUserId(userNickName);
         User user = getUser(userNickName);
 
-        /*UserGroup usergroup = (UserGroup)em.createQuery("SELECT u FROM UserGroup u JOIN u.client c JOIN u.group g WHERE g.groupId=:groupid AND (SELECT FROM c )")
-                .setParameter("userid", userId)
-                .setParameter("groupid", groupId)
-                .getSingleResult();
-*/
+        UserGroup u = (UserGroup)em.createQuery("SELECT g FROM UserGroup g WHERE g.userId=:userid and g.groupId=:groupid")
+                        .setParameter("userid", userId)
+                        .setParameter("groupid", groupId)
+                        .getSingleResult();
+        Access a = getAccess(accessRights);
 
-        //Change info
-       /* usergroup.getAccess().setAccessName(accessRights);
-        int index = usergroup.getClient().indexOf(user);
-        usergroup.getClient().get(index).setNickName(userNickName);
-*/
+        u.setAccess(a);
+
+        em.getTransaction().begin();
+        em.persist(u);
+        em.getTransaction().commit();
     }
+    */
 
     public void deleteMember(int groupId, String userNickName) {
```

Comment:
Please consider passing params of types Group, User/UserGroup. The transformations id->entity, entity->id from outside EJB look ugly.